### PR TITLE
Keep playground structure

### PR DESCRIPTION
**Pull request by miguelangel-dev**
## Issues
- https://github.com/bow-swift/bow/issues/585

## Details
Keep the consistency between the playground structure and render output for `Markdown` and `Jekyll`

### CODE REVIEW

File: project/Component/NefJekyll/Models/RenderJekyllEnvironment.swift [hunk 1]
```diff
@@ -44,12 +44,16 @@ public struct RenderJekyllEnvironment<A> {
...
-            return IO.pure("/\(docs)/\(playground.escapedTitle)/\(page.escapedTitle)/")^
+            return IO.pure("/\(docs)/\(pagePathComponent(playground: playground, page: page))/")^
         default:
             return IO.raiseError(.renderEmpty)^
         }
     }
     
+    static func pagePathComponent(playground: RenderingURL, page: RenderingURL) -> String {
+        "\(playground.escapedTitle)/\(page.escapedTitle)"
+    }
```
Comment (truizlop) [inline project/Component/NefJekyll/Models/RenderJekyllEnvironment.swift]: This looks like it does not belong here. I'd say it should be an instance method of `RenderingURL`. Still, not optimal, but you are using this as a static method from the outside, which should not happen.

File: project/Component/NefJekyll/NefJekyll.swift [hunk 2]
```diff
@@ -76,7 +80,10 @@ public struct Jekyll {
     }
     
     private func writePlayground(playground: RenderingURL, content: PlaygroundOutput, output: URL) -> EnvIO<Environment, RenderError, URL> {
-        content.traverse { info in self.writePage(page: info.page, content: info.output, output: output) }
+        
+        content.traverse { info in self.writePage(pathComponent: Environment.pagePathComponent(playground: playground, page: info.page),
```
Comment (truizlop) [inline project/Component/NefJekyll/NefJekyll.swift]: A method from the environment should not be static.

**Review by truizlop — CHANGES_REQUESTED** [review]

File: project/Component/NefCommon/Models/RenderingURL.swift [hunk 3]
```diff
@@ -10,6 +10,7 @@ public struct RenderingURL {
 
     public var escapedTitle: String {
         title.lowercased().replacingOccurrences(of: "?", with: "-")
+                          .replacingOccurrences(of: "→", with: "")
```
Comment (truizlop) [inline project/Component/NefCommon/Models/RenderingURL.swift]: I have had troubles in the past with titles containing `:`, so it'd be nice to add them here. In fact, we should probably consider replacing any non-alphabetical or non-numeric character to ensure we always have a valid file name.

Comment (miguelangel-dev) [reply project/Component/NefCommon/Models/RenderingURL.swift]: Not sure, why I did not do it before, but it makes sense :)

File: project/nef.xcodeproj/xcshareddata/xcschemes/Markdown.xcscheme [hunk 4]
```diff
@@ -50,6 +50,16 @@
...
+      <CommandLineArguments>
+         <CommandLineArgument
+            argument = "--project ~/Desktop/Documentation.app"
+            isEnabled = "YES">
+         </CommandLineArgument>
+         <CommandLineArgument
+            argument = "--output ~/Desktop/Migue2"
+            isEnabled = "YES">
+         </CommandLineArgument>
+      </CommandLineArguments>
```
Comment (truizlop) [inline project/nef.xcodeproj/xcshareddata/xcschemes/Markdown.xcscheme]: This information should not be committed to the repository, as it is local to your computer.

File: project/Component/NefCommon/Models/RenderingURL.swift [hunk 5]
```diff
@@ -9,8 +9,8 @@ public struct RenderingURL {
     public let title: String
 
     public var escapedTitle: String {
-        title.lowercased().replacingOccurrences(of: "?", with: "-")
-                          .replacingOccurrences(of: " ", with: "-")
+        String(title.lowercased().unicodeScalars.filter { !$0.isAlphanumeric || $0 == " " || $0 == "-" })
```
Comment (truizlop) [inline project/Component/NefCommon/Models/RenderingURL.swift]: Aren't `" "` and `"-"` already checked in the `!$0.isAlphanumeric`?

File: project/Component/NefCommon/Models/RenderingURL.swift [hunk 6]
```diff
@@ -19,6 +19,12 @@ public struct RenderingURL {
     }
 }
 
+extension Unicode.Scalar {
+    var isAlphanumeric: Bool {
+        CharacterSet.alphanumerics.inverted.contains(self)
+    }
```
Comment (truizlop) [inline project/Component/NefCommon/Models/RenderingURL.swift]: This is checking exactly the opposite, as you are inverting the character set.

**Review by truizlop — CHANGES_REQUESTED** [review]

File: project/Component/NefCommon/Models/RenderingURL.swift (hunk 6 above)
Comment (miguelangel-dev) [reply project/Component/NefCommon/Models/RenderingURL.swift]: puf! you are right, good catch!

File: project/Component/NefCommon/Models/RenderingURL.swift (hunk 5 above)
Comment (miguelangel-dev) [reply project/Component/NefCommon/Models/RenderingURL.swift]: It is a good question, in fact like you, I thought these characters are in the non-base characters, but it isn't

FYI: `CharacterSet.alphanumeric` contains (from Apple documentation):
```
A character set containing the characters in Unicode General Categories L*, M*, and N*.
```
and we can check the category M*
1. [Unicode Character Category 'Mark, Spacing Combining](https://www.fileformat.info/info/unicode/category/Mc/index.htm)
2. [Unicode Character Category 'Mark, Enclosing'](https://www.fileformat.info/info/unicode/category/Me/index.htm)
3. [Unicode Character Category 'Mark, Nonspacing'](https://www.fileformat.info/info/unicode/category/Mn/index.htm)

and also, if we check directly to the character set it doesn't return it:
```
CharacterSet.alphanumerics.contains(" ") // false
CharacterSet.alphanumerics.contains("-") // false
```

**Review by truizlop — APPROVED** [review]

**Pull request merged**